pkgs/package-overrides.nix

```
@@ -347,6 +347,8 @@ in
             ];
         in
         ourPatches ++ upstreamPatches;
+
+      buildInputs = attrs.buildInputs ++ (if (lib.versionAtLeast prev.php.version "8.1") then [ pkgs.openssl ] else [ pkgs.openssl_1_1 ]);
```

Maybe something like

```suggestion
      buildInputs =
        let
          replaceOpenssl =
            if lib.versionAtLeast prev.php.version "8.1" then
              lib.id
            else
              pkg:
                if pkg == pkgs.openssl then
                  pkgs.openssl_1_1.overrideAttrs (old: {
                    meta = builtins.removeAttrs old.meta [ "knownVulnerabilities" ];
                  });
                else
                  pkg
        in
        builtins.map replaceOpenssl attrs.buildInputs;
```

---

And then we can remove it from `phps.nix`